src/zkevm_specs/evm/step.py

```diff
@@ -0,0 +1,432 @@
+                assert allocation_revert[1] == True
+                assert allocation_revert[2] == tag
+                if tag == RWTableTag.TxAccessListAccount:
+                    assert allocation_revert[3] == allocations[3]  # tx_id
+                    assert allocation_revert[4] == allocations[4]  # account address
+                    assert allocation_revert[5] == allocations[6]  # revert value
+                elif tag == RWTableTag.TxAccessListStorageSlot:
+                    assert allocation_revert[3] == allocations[3]  # tx_id
+                    assert allocation_revert[4] == allocations[4]  # account address
+                    assert allocation_revert[5] == allocations[5]  # storage slot
+                    assert allocation_revert[6] == allocations[7]  # revert value
+                elif tag == RWTableTag.AccountNonce:
+                    assert allocation_revert[3] == allocations[3]  # account address
+                    assert allocation_revert[4] == allocations[5]  # revert value
+                elif tag == RWTableTag.AccountBalance:
+                    assert allocation_revert[3] == allocations[3]  # account address
+                    assert allocation_revert[4] == allocations[5]  # revert value
+                elif tag == RWTableTag.AccountCodeHash:
+                    assert allocation_revert[3] == allocations[3]  # account address
+                    assert allocation_revert[4] == allocations[5]  # revert value
+                elif tag == RWTableTag.AccountStorage:
+                    assert allocation_revert[3] == allocations[3]  # account address
+                    assert allocation_revert[4] == allocations[4]  # storage slot
+                    assert allocation_revert[5] == allocations[6]  # revert value
+                assert self.tables.rw_lookup(allocation_revert)
+
+                self.state_write_counter_diff += 1
+
+        return allocations[3+len(inputs):]
+
+    def opcode_lookup(self, offset: int = 0) -> Opcode:
+        if self.call.is_create:
+            if self.call.is_root:
+                return Opcode(self.tx_lookup(TxTableTag.Calldata, [
+                    self.call.opcode_source,
+                    self.call.program_counter + offset,
+                ]))
+            else:
+                # TODO: Add offset and verify creation code length when dealing with create
+                return Opcode(self.r_lookup(RWTableTag.Memory, [
+                    self.call.opcode_source,
+                    self.call.program_counter + offset,
+                ])[0])
+        else:
+            return self.bytecode_lookup([
+                self.call.opcode_source,
+                self.call.program_counter + offset,
+            ])
+
+    def stack_pop_lookup(self) -> int:
+        value = self.r_lookup(RWTableTag.Stack, [
+            self.core.call_id,
+            self.call.stack_pointer + self.stack_pointer_diff
+        ])[0]
+        self.stack_pointer_diff += 1
```

Comment:
do we need to move "self.stack_pointer_diff += 1" to the beginning  of current method ?

Reply:
When pushing data to stack, we move (decrease) the `stack_pointer` first, then write the data. So when popping the value, we should do it reversely, read the data first, then move (increase) the `stack_pointer`, right?

Reply:
yeah, agree!